File: generators/hydraNames.py
__all__ = ['hydraNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm4', 'nameGen', 'nm5', 'nm3', 'nm2'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['result', 'type'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
            var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
            var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
            if (var.get('i')<Js(4.0)):
                if (var.get('rnd')<Js(5.0)):
                    while (var.get('rnd5')<Js(7.0)):
                        var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5'))))
            else:
                if (var.get('i')<Js(6.0)):
                    var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                    var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                    while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm5').get(var.get('rnd7'))):
                        var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                    var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm5').get(var.get('rnd7')))+var.get('nm2').get(var.get('rnd6')))+var.get('nm4').get(var.get('rnd5'))))
                else:
                    if (var.get('i')<Js(8.0)):
                        var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                        var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                        while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm5').get(var.get('rnd7'))):
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                        var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm5').get(var.get('rnd7')))+var.get('nm2').get(var.get('rnd6')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5'))))
                    else:
                        var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                        var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                        while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm3').get(var.get('rnd7'))):
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                        var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd7')))+var.get('nm2').get(var.get('rnd6')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('ch'), Js('g'), Js('gh'), Js('j'), Js('jh'), Js('k'), Js('kh'), Js('kr'), Js('m'), Js('mr'), Js('mg'), Js('n'), Js('ng'), Js('nr'), Js('tr'), Js('v'), Js('z'), Js('zh'), Js('zr')]))
var.put('nm2', Js([Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('au'), Js('ou'), Js('ua'), Js('ia'), Js('ai'), Js('ae')]))
var.put('nm3', Js([Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('h'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('gg'), Js('k'), Js('kk'), Js('ll'), Js('m'), Js('mm'), Js('n'), Js('nn'), Js('r'), Js('ss'), Js('v'), Js('z'), Js('zz'), Js('ch'), Js('g'), Js('gz'), Js('gr'), Js('gn'), Js('gg'), Js('h'), Js('hj'), Js('hsh'), Js('hz'), Js('k'), Js('kk'), Js('kh'), Js('ll'), Js('m'), Js('mm'), Js('mz'), Js('n'), Js('ng'), Js('nn'), Js('nz'), Js('nk'), Js('ngr'), Js('r'), Js('rd'), Js('rg'), Js('rz'), Js('sh'), Js('ss'), Js('shg'), Js('shk'), Js('sk'), Js('ssh'), Js('ssk'), Js('sz'), Js('v'), Js('vh'), Js('z'), Js('zr'), Js('zsh'), Js('zg'), Js('zk'), Js('zz')]))
var.put('nm4', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('hs'), Js('hz'), Js('j'), Js('n'), Js('s'), Js('sh'), Js('z')]))
var.put('nm5', Js([Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('k'), Js('m'), Js('n'), Js('r'), Js('v'), Js('z'), Js('g'), Js('gg'), Js('k'), Js('kk'), Js('ll'), Js('m'), Js('mm'), Js('n'), Js('nn'), Js('r'), Js('ss'), Js('v'), Js('z'), Js('zz')]))
pass
pass


# Add lib to the module scope
hydraNames = var.to_python()

File: js/hydraNames.js
var nm1 = ["","","","","","ch","g","gh","j","jh","k","kh","kr","m","mr","mg","n","ng","nr","tr","v","z","zh","zr"];
var nm2 = ["a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","a","e","i","o","u","au","ou","ua","ia","ai","ae"];
var nm3 = ["g","k","m","n","r","v","z","g","h","k","m","n","r","v","z","g","k","m","n","r","v","z","g","k","m","n","r","v","z","g","gg","k","kk","ll","m","mm","n","nn","r","ss","v","z","zz","ch","g","gz","gr","gn","gg","h","hj","hsh","hz","k","kk","kh","ll","m","mm","mz","n","ng","nn","nz","nk","ngr","r","rd","rg","rz","sh","ss","shg","shk","sk","ssh","ssk","sz","v","vh","z","zr","zsh","zg","zk","zz"];
var nm4 = ["","","","","","","","h","hs","hz","j","n","s","sh","z"];
var nm5 = ["g","k","m","n","r","v","z","g","k","m","n","r","v","z","g","k","m","n","r","v","z","g","k","m","n","r","v","z","g","gg","k","kk","ll","m","mm","n","nn","r","ss","v","z","zz"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.floor(Math.random() * nm1.length);
		rnd2 = Math.floor(Math.random() * nm2.length);
		rnd3 = Math.floor(Math.random() * nm3.length);
		rnd4 = Math.floor(Math.random() * nm2.length);
		rnd5 = Math.floor(Math.random() * nm4.length);
		if(i < 4){
			if(rnd < 5){
				while(rnd5 < 7){
					rnd5 = Math.floor(Math.random() * nm4.length);
				}
			}
			names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm4[rnd5];
		}else if(i < 6){
			rnd6 = Math.floor(Math.random() * nm2.length);
			rnd7 = Math.floor(Math.random() * nm5.length);
			while(nm3[rnd3] === nm5[rnd7]){
				rnd7 = Math.floor(Math.random() * nm5.length);
			}
			names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm5[rnd7] + nm2[rnd6] + nm4[rnd5];
		}else if(i < 8){
			rnd6 = Math.floor(Math.random() * nm2.length);
			rnd7 = Math.floor(Math.random() * nm5.length);
			while(nm3[rnd3] === nm5[rnd7]){
				rnd7 = Math.floor(Math.random() * nm5.length);
			}
			names = nm1[rnd] + nm2[rnd2] + nm5[rnd7] + nm2[rnd6] + nm3[rnd3] + nm2[rnd4] + nm4[rnd5];
		}else{
			rnd6 = Math.floor(Math.random() * nm2.length);
			rnd7 = Math.floor(Math.random() * nm3.length);
			while(nm3[rnd3] === nm3[rnd7]){
				rnd7 = Math.floor(Math.random() * nm3.length);
			}
			names = nm1[rnd] + nm2[rnd2] + nm3[rnd7] + nm2[rnd6] + nm3[rnd3] + nm2[rnd4] + nm4[rnd5];
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}